BlazarService/src/main/java/com/hubspot/blazar/util/ModuleDiscovery.java

```diff
@@ -1,69 +1,13 @@
 package com.hubspot.blazar.util;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.dataformat.xml.XmlMapper;
-import com.google.common.base.Optional;
-import com.google.common.base.Preconditions;
-import com.google.inject.Inject;
 import com.hubspot.blazar.base.GitInfo;
 import com.hubspot.blazar.base.Module;
-import org.kohsuke.github.GHContent;
-import org.kohsuke.github.GHRepository;
-import org.kohsuke.github.GHTree;
-import org.kohsuke.github.GHTreeEntry;
-import org.kohsuke.github.GitHub;
+import com.hubspot.blazar.github.GitHubProtos.PushEvent;
 
 import java.io.IOException;
-import java.util.HashSet;
-import java.util.Map;
 import java.util.Set;
 
-public class ModuleDiscovery {
-  private final Map<String, GitHub> gitHubByHost;
-  private final XmlMapper xmlMapper;
-
-  @Inject
-  public ModuleDiscovery(Map<String, GitHub> gitHubByHost, XmlMapper xmlMapper) {
-    this.gitHubByHost = gitHubByHost;
-    this.xmlMapper = xmlMapper;
-  }
-
-  public Set<Module> discover(GitInfo gitInfo) throws IOException {
-    GitHub gitHub = gitHubFor(gitInfo);
-    GHRepository repository = gitHub.getRepository(gitInfo.getFullRepositoryName());
-    GHTree tree = repository.getTreeRecursive(gitInfo.getBranch(), 1);
-
-    Set<String> poms = new HashSet<>();
-    for (GHTreeEntry entry : tree.getTree()) {
-      if (isPom(entry.getPath())) {
-        poms.add(entry.getPath());
-      }
-    }
-
-    Set<Module> modules = new HashSet<>();
-    for (String pom : poms) {
-      GHContent content = repository.getFileContent(pom, gitInfo.getBranch());
-      JsonNode node = xmlMapper.readTree(content.getContent());
-      String artifactId = node.get("artifactId").textValue();
-      final String glob;
-      if ("pom".equals(node.path("packaging").textValue())) {
-        glob = pom;
-      } else {
-        glob = (pom.contains("/") ? pom.substring(0, pom.lastIndexOf('/') + 1) : "") + "**";
-      }
-      modules.add(new Module(Optional.<Integer>absent(), artifactId, pom, glob, true));
-    }
-
-    return modules;
-  }
-
-  private GitHub gitHubFor(GitInfo gitInfo) {
-    String host = gitInfo.getHost();
-
-    return Preconditions.checkNotNull(gitHubByHost.get(host), "No GitHub found for host " + host);
-  }
-
-  private static boolean isPom(String path) {
-    return "pom.xml".equals(path) || path.endsWith("/pom.xml");
-  }
+public interface ModuleDiscovery {
+  boolean shouldRediscover(GitInfo gitInfo, PushEvent pushEvent) throws IOException;
+  Set<Module> discover(GitInfo gitInfo) throws IOException;
```

Comment:
@also this is the interface you would need to implement (you can extend `AbstractModuleDiscovery` to get some helper methods and you can look at `MavenModuleDiscovery` for an example impl)